va/validation-authority.go

```diff
@@ -82,11 +85,13 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 			Type:   core.DNSSECProblem,
 			Detail: dnssecErr.Error(),
 		}
-	} else {
+		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] DNSSEC failure: %s", identifier, err))
+	} else if err != nil {
```

Comment:
Whoops! I mean't to wrap this `if  else` in a `if err != nil` so we would only have to do `challenge.Status = core.StatusInvalid; return challenge, err` once.

That should probably be added to both of the places you've changed `LookupHost` calls as at the moment the challenge isn't being invalidated by these errors and the validation continues because we aren't returning.